src/Query/QueryBuilder.php

```diff
@@ -1259,6 +1267,34 @@ public function resetOrderBy(): self
         return $this;
     }
 
+    /**
+     * Adds a CTE to a select query.
+     *
+     * @param string $name         The name of the CTE.
+     * @param self   $queryBuilder The query builder to use within the CTE.
+     *
+     * @return $this This QueryBuilder instance.
+     */
+    public function with(string $name, self $queryBuilder): self
```

Comment:
```suggestion
    public function with(string $name, string|self $part): self
```

Like with unions, I believe we should allow the CTE to be passed as a string.